packages/coreutils/src/json.ts

```diff
@@ -102,7 +144,11 @@ namespace JSONExt {
   export
   function isArray(value: ReadonlyJSONValue): value is ReadonlyJSONArray;
   export
-  function isArray(value: ReadonlyJSONValue): boolean {
+  function isArray(value: PartialJSONValue): value is PartialJSONArray;
+  export
+  function isArray(value: ReadonlyPartialJSONValue): value is ReadonlyPartialJSONArray;
```

Comment:
This can break third party code which is using the functions as type assertion checks, since it's changing the return type. Instead, let's add overloads for each of these functions.

Reply:
I think the current code only adds overloads, and leaves the return type untouched. It does change the argument type though.

Reply:
Do you have an example that currently works, but will fail after this change? It would make it a lot easier to get this right.

Reply:
Note: With the current code, these narrowings are in effect:
```ts
// Same as before:
let a = [] as JSONValue;
let b = [] as ReadonlyJSONValue;
if (JSONExt.isArray(a)) {
  // a: JSONArray
}
if (JSONExt.isArray(b)) {
  // b: ReadonlyJSONArray
}

// New possibilities:
let c = [] as PartialJSONValue;
let d = [] as ReadonlyPartialJSONValue;
if (JSONExt.isArray(c)) {
  // c: PartialJSONArray
}
if (JSONExt.isArray(d)) {
  // d: ReadonlyPartialJSONArray
}
```